glean-core/android/src/main/java/mozilla/telemetry/glean/private/BooleanMetricType.kt

```diff
@@ -14,16 +14,15 @@ import mozilla.telemetry.glean.rust.RustError
 import mozilla.telemetry.glean.Dispatchers
 import mozilla.telemetry.glean.rust.toBoolean
 
-class BooleanMetricType(
-    disabled: Boolean,
-    category: String,
-    lifetime: Lifetime,
-    name: String,
-    val sendInPings: List<String>
-) {
-    private var handle: Long
+class BooleanMetricType internal constructor(var handle: Long, val sendInPings: List<String>) {
```

Comment:
nit: I'd personally flash out the constructor as the other one, so that it's more visible:

```
class BooleanMetricType internal constructor(
  var..
) {
  constructor(
    disabled: ...
  ) : this(0, sendInPings) {

  }
```

Otherwise it's easy to skip on the internal constructor. Please, also comment somewhere on why that's needed.